_posts/2019-06-05-uncertainty-estimates-tfprobability/uncertainty_tfprobability.Rmd

```diff
@@ -0,0 +1,782 @@
+---
+title: "Adding uncertainty estimates to Keras models with tfprobability"
+description: > 
+   As of today, there is no mainstream road to obtain uncertainty estimates from neural networks. All that can be said this that normally, approaches tend to be Bayesian in spirit, involving some way of putting a prior over model weights. This holds true as well for the method presented in this post: We show how to use tfprobability, the R interface to TensorFlow Probability, to add uncertainty estimates to a Keras model in an elegant and conceptually plausible way.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019uncertaintytfp
+date: 06-05-2019
+categories:
+  - Probability and statistics 
+  - Keras
+  - Concepts
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/uci_both.png
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+
+About six months ago, we showed [how to create a custom wrapper](https://blogs.rstudio.com/tensorflow/posts/2018-11-12-uncertainty_estimates_dropout/) to obtain uncertainty estimates from a Keras network. Today we present a less laborious, as well faster-running, way using [tfprobability](https://rstudio.github.io/tfprobability/), the R wrapper to TensorFlow Probability. Like most posts on this blog, this one won't be short, so let's quickly state what you can expect in return of reading time.
+
+## What to expect from this post
+
+Starting from what _not_ to expect: There won't be a recipe that tells you how _exactly_ to set all parameters involved in order to report the "right" uncertainty measures. But then, what _are_ the "right" uncertainty measures? Unless you happen to work with a method that has no (hyper-)parameters to tweak, there will always be uncertainty about how to report uncertainty.
```

Comment:
_there will always be uncertainty about how to report uncertainty._
Do you mean uncertainty in the first mention, here? Is it something closer to ambiguity/no one right way to do it?

Reply:
I think I wanted it to be some kind of pun, but I think this might not be working correctly in English... the first "uncertainty" should have a different connotation, like "not feeling sure about"..."incertitude", perhaps? Or just: "questions"?

Reply:
went with questions now, think I'm ready to merge then...